lib/ansible/playbook/play.py

```diff
@@ -290,6 +290,10 @@ def get_vars_files(self):
             return [self.vars_files]
         return self.vars_files
 
+    def add_handler(self, handler):
+        if handler.name not in [h.name for h in self.handlers]:
+            self.handlers.append(handler)
```

Comment:
Technically speaking this is a change in functionality.  We have always allowed duplicate handler names.

See https://github.com/ansible/ansible/pull/49249

This also would not allow 2 roles that have the same handler name, but can be addressed using the role name as a namespace:

```
role_name : handler name
```

Maybe it's simple enough to use `.get_name()` instead of `.name` to resolve this.  I'm not sure yet.

See https://github.com/ansible/ansible/blob/cfba6dfe9141fb7f66bf10573012bdae3f5a847a/lib/ansible/plugins/strategy/__init__.py#L370 for where we do this lookup, and allow the `get_name()` version which supports namespacing.

Reply:
Thanks! I was indeed checking the wrong name. I just push a new version that should be better. Regarding the order, I don't think it's a problem since this is just the triggers. Not the list of handlers.